vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -190,6 +189,26 @@ ReactInstanceWin::ReactInstanceWin(
           this,
           options.Properties,
           winrt::make<implementation::ReactNotificationService>(options.Notifications))} {
+  m_whenLoaded.AsFuture()
+      .Then<Mso::Executors::Inline>(
+          [onLoaded = m_options.OnInstanceLoaded, reactContext = m_reactContext](Mso::Maybe<void> &&value) noexcept {
+            if (onLoaded) {
+              onLoaded.Get()->Invoke(reactContext, value.IsError() ? value.TakeError() : Mso::ErrorCode());
```

Comment:
Do we need the weak/strong dance here for `reactcontext`?

Reply:
similar for below.

Reply:
No, we do not need it. The `ReactContext` is a specialized weak pointer to the `ReactInstance`. Using a weak pointer to the `ReactInstance` is so common that in our API we decided to never expose the `ReactInstance`, but rather only give its weak pointer - the `ReactContext`.